scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala

```diff
@@ -309,6 +308,10 @@ class LegacyScanner(input: Input, dialect: Dialect) {
    */
   protected final def fetchToken() {
     offset = charOffset - 1
+
+    if (inStringInterpolation) return fetchStringPart()
+    else if (inXmlLiteral) return fetchXmlPart()
```

Comment:
I did this because both `fetchStringPart` and `fetchXmlPart` were relying on offset to be set. This was not the case in `fetchXmlPart ` before and so we were emitting wrong tokens